server/lib/tuist_web/live/integrations_live.ex

```diff
@@ -0,0 +1,179 @@
+    vcs_connections = vcs_connections(selected_account)
+
+    socket =
+      socket
+      |> assign(selected_tab: "integrations")
+      |> assign(selected_account: selected_account)
+      |> assign(github_app_installation: github_installation)
+      |> assign(vcs_connections: vcs_connections)
+      |> assign(selected_project_id: nil)
+      |> assign(selected_repository_full_handle: nil)
+      |> assign(:head_title, "#{gettext("Integrations")} · #{selected_account.name} · Tuist")
+      |> then(fn socket ->
+        if github_installation do
+          assign_async(socket, :github_repositories, fn ->
+            {:ok, repositories} = GitHubAppInstallations.get_repositories(github_installation)
+            {:ok, %{github_repositories: repositories}}
+          end)
+        else
+          assign(socket, github_repositories: %{ok?: true, result: [], loading: false})
+        end
+      end)
+
+    {:ok, socket}
+  end
+
+  @impl true
+  def handle_event("close-add-connection-modal", _params, socket) do
+    socket = push_event(socket, "close-modal", %{id: "add-connection-modal"})
+
+    {:noreply, socket}
+  end
+
+  @impl true
+  def handle_event("select-project", %{"project_id" => project_id}, socket) do
+    {:noreply, assign(socket, selected_project_id: String.to_integer(project_id))}
+  end
+
+  @impl true
+  def handle_event("select-repository", %{"repository" => repository_full_handle}, socket) do
+    {:noreply, assign(socket, selected_repository_full_handle: repository_full_handle)}
+  end
+
+  @impl true
+  def handle_event(
+        "create-connection",
+        _params,
+        %{
+          assigns:
+            %{
+              selected_repository_full_handle: repository_full_handle,
+              selected_account: selected_account,
+              current_user: current_user
+            } = assigns
+        } = socket
+      ) do
+    project = get_selected_project(assigns)
+    project_id = if project, do: project.id
+
+    attrs = %{
+      project_id: project_id,
+      provider: :github,
+      repository_full_handle: repository_full_handle,
+      created_by_id: current_user.account.id,
+      github_app_installation_id: selected_account.github_app_installation.id
+    }
+
+    {:ok, _connection} = Projects.create_vcs_connection(attrs)
+    vcs_connections = vcs_connections(selected_account, force: true)
+
+    socket =
+      socket
+      |> assign(vcs_connections: vcs_connections)
+      |> assign(selected_project_id: nil)
+      |> assign(selected_repository_full_handle: nil)
+      |> push_event("close-modal", %{id: "add-connection-modal"})
+
+    {:noreply, socket}
+  end
+
+  @impl true
+  def handle_event("delete-connection", %{"connection_id" => connection_id}, %{assigns: assigns} = socket) do
+    %{selected_account: selected_account} = assigns
+
+    {:ok, connection} = Projects.get_vcs_connection(connection_id)
+    {:ok, _} = Projects.delete_vcs_connection(connection)
```

Comment:
Can't remember, but having authorized the socket connection, do we need to authorize these actions too?